swift/ql/test/library-tests/dataflow/flowsources/FlowSourcesInline.ql

```diff
@@ -20,6 +30,19 @@ module FlowSourcesTest implements TestSig {
       value = describe(source)
     )
   }
+
+  predicate hasOptionalResult(Location location, string element, string tag, string value) {
+    // this is not really what the "flowsources" test is about, but sometimes it's helpful to
+    // confirm that taint reaches certain obvious points in the flow source test code.
+    exists(DataFlow::Node n |
+      TaintReachFlow::flowTo(n) and
+      location = n.getLocation() and
+      location.getFile().getBaseName() != "" and
+      element = n.toString() and
+      tag = "tainted" and
+      value = ""
+    )
+  }
```

Comment:
Why not just replace `x` with `sink(x)` in the tests whenever we want to test that taint has reached the value?

Reply:
Yeah, that sounds like the right way to do it, assuming I hadn't hit some problem with that approach...

Reply:
Ah, it's because this is the "flowsources" test and it doesn't have `sink(x)` methods.  Nevertheless adding sinks is probably a better approach than this ad-hoc design.